src/Microsoft.AspNet.Mvc.Core/ActionConstraints/ConsumesAttribute.cs

```diff
@@ -146,11 +147,29 @@ private bool IsApplicable(ActionDescriptor actionDescriptor)
 
         private List<MediaTypeHeaderValue> GetContentTypes(string firstArg, string[] args)
         {
+            var completeArgs = args.ToList();
+            completeArgs.Insert(0, firstArg);
             var contentTypes = new List<MediaTypeHeaderValue>();
-            contentTypes.Add(MediaTypeHeaderValue.Parse(firstArg));
-            foreach (var item in args)
+            foreach (var arg in completeArgs)
             {
-                var contentType = MediaTypeHeaderValue.Parse(item);
+                MediaTypeHeaderValue contentType = null;
+                try
+                {
+                    contentType = MediaTypeHeaderValue.Parse(arg);
+                }
+                catch (FormatException formatException)
```

Comment:
why are we doing this? Shouldn't the exception coming from the runtime be good enough?